llvm-hs-pure/src/LLVM/IRBuilder/Internal/SnocList.hs

```diff
@@ -1,12 +1,23 @@
+{-# LANGUAGE CPP #-}
 module LLVM.IRBuilder.Internal.SnocList where
 
+#if MIN_VERSION_base(4,11,0)
 import LLVM.Prelude
+#else
+import Data.Semigroup (Semigroup(..))
+import LLVM.Prelude hiding ((<>))
```

Comment:
I briefly debated whether to stuff `Semigroup` into `LLVM.Prelude`, but I decided against it and went with this approach, which required fewer changes. Let me know if you'd prefer it the other way around.